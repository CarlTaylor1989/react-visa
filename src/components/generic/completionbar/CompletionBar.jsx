import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-redux-i18n';
import NumberFormat from 'react-number-format';

import './CompletionBar.scss';

const CompletionBar = (props) => {
  const {
    amount,
    displayValue,
    hiddenTextField,
    percent,
    total
  } = props;
  const style = { width: `${percent}%` };
  return (
    <div className="completionBar">
      <div className="inner" style={style} />
      <div className="completion">
        <NumberFormat
          value={amount}
          displayType="text"
          thousandSeparator=","
        />
        <Translate value="generic.completionProgressDivider" />
        {displayValue ? (
          <NumberFormat
            value={total}
            displayType="text"
            thousandSeparator=","
          />
        ) : (
          <Translate value={hiddenTextField} />
        )}
      </div>
    </div>
  );
};

CompletionBar.propTypes = {
  amount: PropTypes.number.isRequired,
  displayValue: PropTypes.bool,
  hiddenTextField: PropTypes.string,
  percent: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired
};

CompletionBar.defaultProps = {
  displayValue: true,
  hiddenTextField: ''
};

export default CompletionBar;
