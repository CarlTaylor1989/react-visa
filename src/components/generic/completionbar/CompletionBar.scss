@import '../../../styles/mixins';
@import '../../../styles/variables';

.completionBar {
  @include border-radius(17px);
  background-color: $completion-bar-colour;
  height: 30px;
  padding: 3px;
  position: relative;
  width: 100%;

  .inner {
    @include border-radius(17px);
    @include multi-left-right-gradient(rgba($colour-white, .15), rgba($colour-white, .8));
    height: 30px;
    transition: width .5s ease-in;
  }

  .completion {
    font-size: 20px;
    font-weight: 700;
    left: 20px;
    line-height: 30px;
    margin-top: -15px;
    position: absolute;
    top: 50%;
  }
}
