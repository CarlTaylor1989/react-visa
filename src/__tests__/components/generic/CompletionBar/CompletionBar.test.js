/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import CompletionBar from '../../../../components/generic/completionbar/CompletionBar';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    percent: 75,
    amount: 150,
    total: 200
  };
});

test('renders the CompletionBar component with the total value shown', () => {
  const wrapper = shallow(<CompletionBar {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the CompletionBar component with a hidden text', () => {
  mockProps.displayValue = false;
  mockProps.hiddenTextField = 'generic.dashboard.hiddenValue';
  const wrapper = shallow(<CompletionBar {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
