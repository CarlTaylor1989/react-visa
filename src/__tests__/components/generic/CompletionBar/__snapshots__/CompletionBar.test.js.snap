// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CompletionBar component with a hidden text 1`] = `
<div
  className="completionBar"
>
  <div
    className="inner"
    style={
      Object {
        "width": "75%",
      }
    }
  />
  <div
    className="completion"
  >
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={150}
    />
    <Translate
      tag="span"
      value="generic.completionProgressDivider"
    />
    <Translate
      tag="span"
      value="generic.dashboard.hiddenValue"
    />
  </div>
</div>
`;

exports[`renders the CompletionBar component with the total value shown 1`] = `
<div
  className="completionBar"
>
  <div
    className="inner"
    style={
      Object {
        "width": "75%",
      }
    }
  />
  <div
    className="completion"
  >
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={150}
    />
    <Translate
      tag="span"
      value="generic.completionProgressDivider"
    />
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix=""
      suffix=""
      thousandSeparator=","
      thousandsGroupStyle="thousand"
      type="text"
      value={200}
    />
  </div>
</div>
`;
